src/Nancy/ErrorHandling/DefaultStatusCodeHandler.cs

```diff
@@ -62,47 +66,60 @@ public void Handle(HttpStatusCode statusCode, NancyContext context)
                 return;
             }
 
-            string errorPage;
-
-            if (!this.errorPages.TryGetValue(statusCode, out errorPage))
+            if (!this.errorMessages.ContainsKey(statusCode) || !this.errorPages.ContainsKey(statusCode))
             {
                 return;
             }
 
-            if (String.IsNullOrEmpty(errorPage))
+            var result = new DefaultStatusCodeHandlerResult
             {
+                Details = StaticConfiguration.DisableErrorTraces ? DisableErrorTracesTrueMessage : context.GetExceptionDetails(),
+                Message = this.errorMessages[statusCode],
+                StatusCode = statusCode
+            };
+
+            try
+            {
+                context.Response = this.responseNegotiator.NegotiateResponse(result, context);
+                context.Response.StatusCode = statusCode;
                 return;
             }
-
-            Func<HttpStatusCode, NancyContext, string, string> expansionDelegate;
-            if (this.expansionDelegates.TryGetValue(statusCode, out expansionDelegate))
+            catch (ViewNotFoundException)

```

Comment:
Well, an exception for control is a red flag of course. I haven't looked closely enough to see if there are alternatives.